commons/src/main/java/io/aiven/kafka/tieredstorage/commons/storage/s3/S3StorageConfig.java

```diff
@@ -137,6 +147,10 @@ AmazonS3 s3Client() {
                 new AwsClientBuilder.EndpointConfiguration(s3ServiceEndpoint, region);
             s3ClientBuilder.withEndpointConfiguration(endpointConfiguration);
         }
+        final Boolean pathStyleAccessEnabled = pathStyleAccessEnabled();
+        if (Objects.nonNull(pathStyleAccessEnabled)) {
```

Comment:
Nit, but we need to agree on this at some point:
`x != null` is much more pleasant to read than `Objects.nonNull(x)`. `Objects.nonNull` is meant to be a ready-made predicate for stream filtering and the like, not to replace the normal comparison.

Reply:
Make sense. I brought it as I saw it adopted in other packages, but your proposal to when to use it is sound.